src/main/java/org/alfresco/heartbeat/RenditionsDataCollector.java

```diff
@@ -0,0 +1,131 @@
+ *      </ul>
+ *  </li>
+ * </ul>
+ *
+ * @author eknizat
+ */
+public class RenditionsDataCollector extends HBBaseDataCollector implements InitializingBean
+{
+    /** The logger. */
+    private static final Log logger = LogFactory.getLog(RenditionsDataCollector.class);
+
+    private DescriptorDAO currentRepoDescriptorDAO;
+
+    // Map keyed on rendition id to a Map keyed on source mimetypes to a count of the number of times it has been requested.
+    private final Map<ThumbnailDefinition, Map<String, AtomicInteger>> renditionRequests = new ConcurrentHashMap<>();
+
+    public RenditionsDataCollector(String collectorId, String collectorVersion, String cronExpression,
+                                    HeartBeatJobScheduler hbJobScheduler)
+    {
+        super(collectorId, collectorVersion, cronExpression, hbJobScheduler);
+    }
+
+    public void setCurrentRepoDescriptorDAO(DescriptorDAO currentRepoDescriptorDAO)
+    {
+        this.currentRepoDescriptorDAO = currentRepoDescriptorDAO;
+    }
+
+    @Override
+    public void afterPropertiesSet() throws Exception
+    {
+        PropertyCheck.mandatory(this, "currentRepoDescriptorDAO", currentRepoDescriptorDAO);
+    }
+
+    public void recordRenditionRequest(ThumbnailDefinition rendition, String sourceMimetype)
+    {
+        // Increment the count of renditions. Atomically creates missing parts of the Map structures.
+        renditionRequests.computeIfAbsent(rendition,
+                k -> new ConcurrentHashMap<>()).computeIfAbsent(sourceMimetype,
+                k -> new AtomicInteger()).incrementAndGet();
+    }
+
+    @Override
+    public List<HBData> collectData()
+    {
+        List<HBData> collectedData = new LinkedList<>();
+
+        String systemId = this.currentRepoDescriptorDAO.getDescriptor().getId();
+        String collectorId = this.getCollectorId();
+        String collectorVersion = this.getCollectorVersion();
+        Date timestamp = new Date();
+
+        // We don't mind if new renditions are added while we iterate, as we will pick them up next time.
+        for (ThumbnailDefinition rendition : renditionRequests.keySet())
+        {
+            String renditionName = rendition.getName();
+            String targetMimetype = rendition.getMimetype();
+            for (Map.Entry<String, AtomicInteger> entry: renditionRequests.remove(rendition).entrySet())
+            {
+                String sourceMimetype = entry.getKey();
+                AtomicInteger count = entry.getValue();
+
+                Map<String, Object> values = new HashMap<>();
+                values.put("rendition", renditionName);
+                values.put("count", count.intValue());
+                values.put("sourceMimetype", sourceMimetype);
+                values.put("targetMimetype", targetMimetype);
+
+                collectedData.add(new HBData(systemId, collectorId, collectorVersion, timestamp, values));
```

Comment:
Why create different entries for each rendition and not a list of renditions with their respective details?
Will this be easier to search for/visualize in Kibana? How many different renditions are expected?

Reply:
I did think about that myself and even wrote a slack message to you but did not send it.
In the end, I decided it would be simpler to be able to combine results in Kibana from different nodes and days if the data was flattened (denormalised) out at this point. It is very likely that different nodes would have different sets of source mimetypes which would make summing the counts harder to do.